### PR TITLE
(Fixes #62) Adds role and resourceRole to EmrCluster types

### LINKED ISSUE #62: Support role and resourceRole on EmrCluster

### PR DESCRIPTION
**Pull request by sethyates**

### CODE REVIEW

**Comment by ysalomon** [comment]
LGTM

File: src/main/scala/com/krux/hyperion/resource/ActionOnResourceFailure.scala [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package com.krux.hyperion.resource
+
+trait ActionOnResourceFailure
+
+case object RetryAllOnResourceFailure {
```
Comment (realstraw) [inline src/main/scala/com/krux/hyperion/resource/ActionOnResourceFailure.scala]: Shouldn't this extend the above trait?

File: src/main/scala/com/krux/hyperion/resource/ActionOnResourceFailure.scala [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+  override def toString: String = "retryall"
+}
+
+case object RetryNoneOnResourceFailure {
```
Comment (realstraw) [inline src/main/scala/com/krux/hyperion/resource/ActionOnResourceFailure.scala]: Shouldn't this extend the above trait?

File: src/main/scala/com/krux/hyperion/resource/ActionOnTaskFailure.scala [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.krux.hyperion.resource
+
+trait ActionOnTaskFailure
+
+case object ContinueOnTaskFailure {
```
Comment (realstraw) [inline src/main/scala/com/krux/hyperion/resource/ActionOnTaskFailure.scala]: Shouldn't this extend the above trait?

File: src/main/scala/com/krux/hyperion/resource/ActionOnTaskFailure.scala [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+  override def toString: String = "continue"
+}
+
+case object TerminateOnTaskFailure {
```
Comment (realstraw) [inline src/main/scala/com/krux/hyperion/resource/ActionOnTaskFailure.scala]: Shouldn't this extend the above trait?

File: src/main/scala/com/krux/hyperion/resource/ActionOnResourceFailure.scala (hunk 1 above)
Comment (sethyates) [inline src/main/scala/com/krux/hyperion/resource/ActionOnResourceFailure.scala]: Yes.  Added.

**Pull request closed without merging**